codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/route53/StripHostedZoneURLPathMiddlewareRenderable.kt

```diff
@@ -21,7 +20,6 @@ class StripHostedZoneUrlPathMiddlewareRenderable(
     override val position = MiddlewarePosition.AFTER
 
     override fun render(ctx: ProtocolGenerator.GenerationContext, writer: SwiftWriter, op: OperationShape, operationStackName: String) {
-        val inputShapeName = MiddlewareShapeUtils.inputSymbol(symbolProvider, model, op).name
-        writer.write("$operationStackName.${middlewareStep.stringValue()}.intercept(position: ${position.stringValue()}, middleware: ${inputShapeName}StripHostedZoneMiddleware())")
+        writer.write("$operationStackName.${middlewareStep.stringValue()}.intercept(position: ${position.stringValue()}, middleware: Route53TrimHostedZoneMiddleware(\\.hostedZoneId))")
```

Comment:
The reusable middleware, initialized with a key path, is inserted into the operation stack instead of a code-generated middleware.